pkg/controller/scaledobject/hpa.go

```diff
@@ -53,7 +53,7 @@ func (r *ReconcileScaledObject) newHPAForScaledObject(logger logr.Logger, scaled
 
 	var behavior *autoscalingv2beta2.HorizontalPodAutoscalerBehavior
 	if r.kubeVersion.MinorVersion >= 18 {
-		behavior = scaledObject.Spec.Behavior
+		behavior = scaledObject.Spec.Advanced.HorizontalPodAutoscalerConfig.Behavior
```

Comment:
`Advanced` and `HorizontalPodAutoscalerConfig` could be nil, right? should the condition be updated to `&& scaledObject.Spec.Advanced != nil && scaledObject.Spec.Advanced.HorizontalPodAutoscalerConfig != nil`?

Also in other places

Reply:
valid point. Have made the change